File: 0383-ransom-note/0383-ransom-note.kt
class Solution {
    fun canConstruct(ransomNote: String, magazine: String): Boolean {
        val chars = IntArray(26)

        for (c in magazine) {
            chars[c - 'a']++
        }

        for (c in ransomNote) {
            if (chars[c - 'a'] == 0) return false

            chars[c - 'a']--
        }

        return true
    }
}

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        count = {}

        for c in magazine:
            count[c] = count.get(c, 0) + 1
        
        for c in ransomNote:
            if count.get(c, 0) <= 0:
                return False
            
            count[c] = count.get(c, 0) - 1

        return True
        